

```java
public class Test {
    public static void main(String[] args) {
         int i =1;
         if(i==1){
             throw new InvalidValueException("自定义异常抛出");
         }
    }

    /**
     * 自定义的异常类需要继承RuntimeException，并提供多个构造方法
     */
    static class InvalidValueException extends RuntimeException {
        public InvalidValueException() {
            super();
        }
        public InvalidValueException(String message, Throwable cause) {
            super(message, cause);
        }
        public InvalidValueException(String message) {
            super(message);
        }
        public InvalidValueException(Throwable cause) {
            super(cause);
        }
    }

}
```
